frank/io.py

```diff
@@ -0,0 +1,96 @@
+# Frankenstein: 1D disc brightness profile reconstruction from Fourier data
+# using non-parametric Gaussian Processes
+#
+# Copyright (C) 2019-2020  R. Booth, J. Jennings, M. Tazzari
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>
+#
+"""This module has functions that read in datafiles and save fit results to
+   file.
+"""
+
+import numpy as np
+
+def load_uvtable(data_file):
+    """
+    Read in a UVTable with data to be fit.
+
+    Parameters
+    ----------
+    data_file : string
+          UVTable with columns: u [lambda]  v [lambda]  Re(V) [Jy]  Im(V) [Jy]
+                                Weight [Jy^-2]
+
+    Returns
+    -------
+    u, v : array, unit = :math:`\\lambda`
+          u and v coordinates of observations
+    vis : array, unit = Jy
+          Real component of observed visibilities
+    weights : array, unit = Jy^-2
+          Weights assigned to observed visibilities, of the form
+          :math:`1 / \\sigma^2`
+    """
+
+    u, v, vis, weights = np.genfromtxt(data_file).T
+    # TODO: add other file types to accept (.npy, .npz)
+    # TODO: allow other column orders in UVTable
+    # TODO: (optionally) convert u, v from [m] to [lambda]?
+
+    return u, v, vis, weights
+
+
+def save_fit(model, u, v, vis, weights, sol):
```

Comment:
I don't like this function being here because io.py should not depend on the model parameters structure. This needs re-designing.

Reply:
It should be dependent on the model parameters, because parameters like `save_dir` are used to save fit outputs. If the structure of the parameter file `default_parameters.json` is changed, this won't work. But in that case `fit.py` won't work either. We don't need to allow users to change the keywords in `default_parameters.json`; they were chosen carefully.

Reply:
Based on our discussion today, I'll restructure to take in only the needed params.

Reply:
I've done this. Resolving.